lib/trace.js

```diff
@@ -461,14 +461,16 @@ Trace.prototype.getLoadRecord = function(canonical, excludeURLs, parentStack) {
             if (typeof sourceMap == 'string')
               sourceMap = load.metadata.sourceMap = JSON.parse(sourceMap);
 
-            var originalName = load.name.split('!')[0];
-            
-            // force set the filename of the original file
-            sourceMap.file = originalName + '!transpiled';
-
-            // force set the sources list if only one source
-            if (!sourceMap.sources || sourceMap.sources.length <= 1)
-              sourceMap.sources = [originalName];
+            if (load.name.indexOf('!') != -1) {
```

Comment:
Could you clarify how this workaround helps here? Are you only using input source maps with plugins loaded with `!` syntax? If so it may be too specific a workaround unfortunately?

Perhaps we can just change this code to be non-damaging rather, only setting `file` and `sources` properties that are not already set.